pkg/server/backend_manager.go

```diff
@@ -36,40 +36,66 @@ import (
 	"sigs.k8s.io/apiserver-network-proxy/proto/header"
 )
 
-type ProxyStrategy string
+type ProxyStrategy int
 
 const (
 	// With this strategy the Proxy Server will randomly pick a backend from
 	// the current healthy backends to establish the tunnel over which to
 	// forward requests.
-	ProxyStrategyDefault ProxyStrategy = "default"
+	ProxyStrategyDefault ProxyStrategy = iota + 1
 	// With this strategy the Proxy Server will pick a backend that has the same
 	// associated host as the request.Host to establish the tunnel.
-	ProxyStrategyDestHost ProxyStrategy = "destHost"
-
+	ProxyStrategyDestHost
 	// ProxyStrategyDefaultRoute will only forward traffic to agents that have explicity advertised
 	// they serve the default route through an agent identifier. Typically used in combination with destHost
-	ProxyStrategyDefaultRoute ProxyStrategy = "defaultRoute"
+	ProxyStrategyDefaultRoute
 )
 
+func (ps ProxyStrategy) String() string {
+	switch ps {
+	case ProxyStrategyDefault:
+		return "default"
+	case ProxyStrategyDestHost:
+		return "destHost"
+	case ProxyStrategyDefaultRoute:
+		return "defaultRoute"
+	}
+	panic("unhandled ProxyStrategy String()")
```

Comment:
If we're going to panic we need to have a very clear error/log message first. Some clue about what ps is set to is pretty essential.

Reply:
Done.